Source/WebCore/workers/WorkerOrWorkletScriptController.cpp

```diff
@@ -285,10 +285,13 @@ bool WorkerOrWorkletScriptController::loadModuleSynchronously(WorkerScriptFetche
 
     Ref protector { scriptFetcher };
     {
-        auto& promise = JSExecState::loadModule(globalObject, sourceCode.jsSourceCode(), JSC::JSScriptFetcher::create(vm, { &scriptFetcher }));
+        auto* promise = JSExecState::loadModule(globalObject, sourceCode.jsSourceCode(), JSC::JSScriptFetcher::create(vm, { &scriptFetcher }));
         scope.assertNoExceptionExceptTermination();
         RETURN_IF_EXCEPTION(scope, false);
 
+        if (UNLIKELY(!promise))
```

Comment:
Shouldn't the RETURN_IF_EXCEPTION handle this?

Reply:
Discussed offline, we'll turn this into a RELEASE_ASSERT in `JSModuleLoader::loadModule` so we ensure at the source that `nullptr` reaching here is 1:1 with an exception having been thrown.

Reply:
Ah, that should be asserted in `jsCast` anyway - so we truly don't need the second check here.

Reply:
Removed this check in latest push.